src/main/java/picard/sam/util/ReadNameParser.java

```diff
@@ -128,6 +132,25 @@ public boolean addLocationInformation(final String readName, final PhysicalLocat
         }
     }
 
+    public boolean addLocationInformation(final String readName, final PhysicalLocation loc){
+        if (!locationMap.containsKey(readName)) {
```

Comment:
since you are going to have both `if` and `else`it would be easier to read the code if you removed the `!` and switched the cases